ui/app/src/nh-config/create-dimension-form.ts

```diff
@@ -0,0 +1,382 @@
+import { property, query, state } from "lit/decorators.js";
+
+const MIN_RANGE = -1_000_000;
+const MAX_RANGE = 1_000_000;
+
+export default class CreateDimension extends NHComponentShoelace {
+  @property()
+  sensemakerStore!: SensemakerStore;
+  
+  @property()
+  dimensionType!: "input" | "output";
+
+  @property()
+  methodComputation!: "AVG" | "SUM";
+  
+  @contextProvided({ context: matrixContext, subscribe: true })
+  _matrixStore!: MatrixStore;
+
+  @contextProvided({ context: weGroupContext, subscribe: true })
+  weGroupId!: DnaHash;
+
+  _dimensionSchema = object({
+    name: string().min(1, "Must be at least 1 characters").required(),
+    computed: boolean().required(),
+  });
+  
+  _currentMinRange : number = 0;
+  _dimensionRangeSchema = () => object({
+    min: number().min(MIN_RANGE, "Must be at least " + MIN_RANGE).required(),
+    max: number().min(this._currentMinRange + 1, "Must be > " + this._currentMinRange).max(MAX_RANGE, "Must be at most " + MAX_RANGE),
+  });
+  @property() // Only needed when an output dimension range is being computed
+  inputRange!: Range & { range_eh: EntryHash }
+  @property() // Only needed when an output dimension range is being computed
+  computationMethod!: "AVG" | "SUM";
+
+  @state()
+  _dimensionRange: Range = { name: "", kind: { Integer: {
+    min: 0,
+    max: 1,
+  }} as RangeKind };
+  @state()
+  _dimension: Partial<Dimension> = { name: "", computed: this.dimensionType == "output", range_eh: undefined };
+
+  @query("nh-button")
+  submitBtn!: NHButton;
+
+  resetInputErrorLabels(inputs: NodeListOf<any>) {
+    inputs.forEach(input => {
+      input.helpText = "";
+      input.nextElementSibling.style.opacity = 0;
+      input.nextElementSibling.style.visibility = 'hidden';
+    });
+  }
+
+  validateIfUntouched(inputs: NodeListOf<any>) {
+    let existsUntouched = false;
+    inputs.forEach((input) => {
+      // Just validate text field for an input dimension as range will be calculated
+      if(this.dimensionType == "output" && input.name !== "dimension-name") return
+
+      if(input.dataset.touched !== "1" && input.required) {
+        this.handleValidationError.call(this, { path: input.name, err: 'untouched'})
+        existsUntouched = true;
+      }
+    });
+    return existsUntouched
+  }
+
+  async resetForm() {
+    this._dimension = { name: "", computed: false, range_eh: undefined };
+    this._dimensionRange = { name: "", kind: { Integer: {
+      min: 0,
+      max: 0,
+    }} as RangeKind };
+    (this.renderRoot.querySelector('sl-input') as any).value = '';
+    delete (this.renderRoot.querySelector('sl-input') as any).dataset.touched;
+    
+    (this.renderRoot.querySelectorAll('sl-range') as any).forEach(range => {
+      range.value = 0;
+      delete range.dataset.touched;
+    })
+
+    this.submitBtn.loading = false;
+    await this.updateComplete
+  }
+
+  getSumComputationRange(min: number, max: number) : RangeKind {     
```

Comment:
I see that we might need to make some changes here because it requires knowledge of the internals of the numeric types in Javascript and WASM. All Javascript `Number` instances are actually floating point numbers (IEEE 754 standard doubles, to be specific). There's also the `BigInt` type, but that's not necessary to get into right now.

The `Number` type actually has a few constants on it that could be useful in our case, but they actually aren't useful:

* `Number.MIN_VALUE`
* `Number.MAX_VALUE`
* `Number.MIN_SAFE_INTEGER`
* `Number.MAX_SAFE_INTEGER`

The first two `MIN_VALUE` and `MAX_VALUE` would be super useful to us in a double float case, because any number below or above these values is `Number.NEGITIVE_INFINITY` or `Number.POSITIVE_INFINITY`. However, in our WASM code, we are specifically using a `f32`, a single float. This means these specific maximum and minimum values don't apply. What values apply? Well, let's [look it up on Wikipedia](https://en.wikipedia.org/wiki/IEEE_754#Basic_and_interchange_formats), since I don't keep these specific values in my head:
* MAX_F32_VALUE = 3.40 * 10^38
* MIN_F32_VALUE = 1.18 * 10^−38

The other two are the represent the minimum and maximum integer values a double float can represent before truncation errors start to occur. Those are not useful for the type of integer we are using in WASM. We are using `u32`, so we can't represent negative integer values. So our possible values are:
* MAX_U32_VALUE = 4294967295
* MIN_U32_VALUE = 0

However, this could potentially be problematic as [WASM leaves the interpretation of the two `i32` and `i64` values it supplies](https://webassembly.github.io/spec/core/syntax/types.html) up to the code using the type. So we need to be sure that the same semantics on the values are enforced in both the TypeScript and WASM code.

If we were enforcing the signing bit, then our values would be:
* MAX_I32_VALUE = 2147483647
* MIN_I32_VALUE = -2147483648

In either case, we can clearly see that these are well below the `*_SAFE_INTEGER` values of JavaScript's `Number`, which would not be the case if using `i64` or `u64` values (where we would need BigInt).

@pospi @weswalla Tagging you in case you want to read this comment and learn more about numeric types.

For more information on numeric types:
* https://2ality.com/2012/04/number-encoding.html
* https://2ality.com/2014/02/javascript-integers.html
* https://2ality.com/2013/10/safe-integers.html
* https://2ality.com/2017/03/es-integer.html
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
* https://doc.rust-lang.org/beta/reference/types/numeric.html
* https://webassembly.github.io/spec/core/syntax/types.html